app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -1,56 +1,137 @@
+            it?.let { it1 -> emailField.setValid(it1)
+                emailField.setError(emailField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.passwordChange.observe(viewLifecycleOwner){
+            val passwordField = viewBinding.etPassword
+            it?.let { it1 -> passwordField.setValid(it1)
+                passwordField.setError(passwordField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.areFieldsValid.observe(viewLifecycleOwner){
+            it.let {  viewBinding.btnLogin.isEnabled = it == true }
+        }
+    }
+
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+
+        etEmail.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail,
+                it
+            )
+        })
+        etPassword.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword,
+                it
+            )
+        })
+    }
+
+    override fun onTextChanged(editable: Editable, taskyAppcompatEditText: TaskyAppCompatEditText) {
+        if (taskyAppcompatEditText.id == R.id.et_email){
+            viewModel.emailChange(editable.toString() as CharSequence)
+        }
+
+
+        if(taskyAppcompatEditText.id == R.id.et_password){
+            viewModel.passwordChange(editable.toString() as CharSequence)
+        }
+            viewModel.areFieldsValid()
+    }
+
+    private fun removeFieldValidations(loginFragment: LoginFragment?) {
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+
+        etEmail.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail,
+                it
+            )
+        })
+
+        etPassword.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword,
+                it
+            )
+        })
+    }
+
     private fun openRegistrationFragment() {
         (activity as EntryActivity).setTitle((activity as EntryActivity).getString(R.string.create_your_account))
 
-        setFragmentManager(activity?.supportFragmentManager!!)
+        setFragmentManager(activity?.supportFragmentManager as FragmentManager)
+        removeFragment(this)
+
         val registrationFragment = RegistrationFragment.getInstance()
         inflateFragment(registrationFragment, R.id.fragment_container)
     }
 
+    override fun onResume() {
+        super.onResume()
+        setFieldValidations(this)
+    }
+    override fun onDestroyView() {
+        super.onDestroyView()
+        removeFieldValidations(this)
+    }
+
     companion object {
+        private var loginFragment : LoginFragment? = null
```

Comment:
Make it lateinit, so you avoid nullability since you are always creating the instance

So:
``` 
private lateinit var loginFragment: LoginFragment
```

Reply:
Done